### PR TITLE
First release of G-man API

### PR DESCRIPTION
**Pull request by dingus9**

### CODE REVIEW

**Comment by dingus9** [comment]
Start the dev server 
```
cd piedpiper-gman-faas/
> python index.py
```

Tests are coming along but not there yet...

Example curl calls
```
curl -v  http://localhost:8080/gman -XPOST -H 'Content-Type: application/json' -d '{"status": "failed", "message": "testing", "run_id": "aafbd4b1d-e3da-4a1d-bc0c-f47254164ef4", "caller": "test", "project": "test proj"}'

curl -v  http://localhost:8080/gman/193ce180-34c6-4868-8b4e-7a5a598b1a97 -XPUT -H 'Content-Type: application/json' -d '{"status": "info", "message": "testing", "run_id": "aafbd4b1d-e3da-4a1d-bc0c-f47254164ef4", "caller": "test", "project": "testing"}'

curl -v  http://localhost:8080/gman/193ce180-34c6-4868-8b4e-7a5a598b1a97/events
```

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+                   'Environment :: Console',
+                   'Operating System :: OS Independent',
+                   'Programming Language :: Python'],
+      author='Nick Shobe',
+      author_email='nickshobe@gmail.com',
+      license='MIT License',
+      packages=find_packages(),
+      include_package_data=True,
+      zip_safe=False,
+      requires=['flask-restful',
```
Comment (ragingpastry) [inline setup.py]: I believe this needs to be "install_requires". distutils will freak out about the '-' in the package name

**Comment by dingus9** [comment]
```
Name                             Stmts   Miss  Cover   Missing
--------------------------------------------------------------
piedpiper_gman/__init__.py           0      0   100%
piedpiper_gman/app.py               16      6    62%   22-29
piedpiper_gman/artman.py             0      0   100%
piedpiper_gman/config.py             7      0   100%
piedpiper_gman/gman.py              62     18    71%   16-32, 37, 46-47, 50, 59, 71, 91
piedpiper_gman/orm/__init__.py       0      0   100%
piedpiper_gman/orm/models.py        33      0   100%
--------------------------------------------------------------
TOTAL                              118     24    80%
```

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+__pycache__
```
Comment (ragingpastry) [inline .gitignore]: What are your thoughts on expanding this to include https://gitignore.io/api/python ? For the FaaS functions and PiCli I've been trying to use a standard gitignore

File: piedpiper_gman/gman.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        for disallowed in ('task_id', 'timestamp', 'thread_id'):
+            if disallowed in raw:
+                errors['errors'].setdefault(disallowed, []).append(
+                    'may not be specified on a post/create')
+
+        if len(errors['errors']):
+            return errors, 422
+
+        # strftime('%Y-%m-%d %H:%M')
+        # raw['timestamp'] == iso8601 "2019-05-10T13:59:37.815929+00:00"
```
Comment (ragingpastry) [inline piedpiper_gman/gman.py]: Should these be removed?

File: piedpiper_gman/orm/models.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+from peewee import (DatabaseProxy,
+                    DateTimeField,
+                    ForeignKeyField,
+                    Model,
+                    SqliteDatabase,
+                    TextField)
+
+# gman schema: task_id(guid), timestamp(datetime),
+#   status[started, completed, failed, info], caller,
```
Comment (ragingpastry) [inline piedpiper_gman/orm/models.py]: Here too

File: tests/test_gman.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
+import uuid
+
+import pytest
+
+from piedpiper_gman.gman import GMan
+
+# from piedpiper_gman.orm.models import (Task, TaskEvent,
```
Comment (ragingpastry) [inline tests/test_gman.py]: Here too

**Comment by ragingpastry** [comment]
Can you squash up some of these commits and change the PR title? Should be good to go after that !

**Review by ragingpastry — APPROVED** [review]
LGTM!

**Pull request merged**